### PR TITLE
Use expressions

### PR DESCRIPTION
**Pull request by lpil**

### CODE REVIEW

File: concepts/use-expressions/about.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# About
+
+In Gleam it is common to write and use higher order functions, that is functions that take other functions as arguments. Sometimes when using many higher order functions at once the code can become difficult to read, with many layers of indentation.
+
+For example, here is a function that calls several functions that return `Result(Int, Nil)`, and sums the values if all three are successful.
```
Comment (giacomocavalieri) [inline concepts/use-expressions/about.md:5]:
```suggestion
For example, here is a function that calls several functions that return `Result(Int, Nil)`, and sums the values if all four are successful.
```

File: concepts/use-expressions/about.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+pub fn main() -> Result(Int, Nil) {
+  use a <- result.try(function1())
+  use b <- result.try(function2())
+  use c <- result.try(function3())
+  use d <- result.try(function4())
+  Ok(a + b + c + d)
+}
+```
+
+A `use` expression collects all the following statements in the block into and passes it as a callback function as the final argument to the function call. The variables between the `use` keyword and the `<-` symbol are the names of the arguments that will be passed to the callback function.
```
Comment (giacomocavalieri) [inline concepts/use-expressions/about.md:37]: "Into" may be a typo?
```suggestion
A `use` expression collects all the following statements in the block and passes it as a callback function as the final argument to the function call. The variables between the `use` keyword and the `<-` symbol are the names of the arguments that will be passed to the callback function.
```

File: exercises/concept/expert-experiments/.docs/instructions.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Instructions
+
+Daphne has been working on a system to run and record the results of her experiments. Some of the code has become a bit verbose and repetitive, so she's asked you to write some `use` expressions to help clean it up.
+
+## 1. Define the `with_retry` function
+
+Sometimes experiments can fail due to a one-off mistake, so if an experiment fails Daphne want to retry it again to see if it works the second time.
```
Comment (giacomocavalieri) [inline exercises/concept/expert-experiments/.docs/instructions.md:7]:
```suggestion
Sometimes experiments can fail due to a one-off mistake, so if an experiment fails Daphne wants to retry it again to see if it works the second time.
```

**Comment by giacomocavalieri** [comment]
Awesome! This will close #276, right?

**Comment by lpil** [comment]
Yes!!

**Pull request merged**